### DISCUSSION TITLE
yarn issue during self hosted (vm) installation

### DISCUSSION THREAD

**Discussion by jchang69** [outline/outline#7469] [Self Hosting]
Hello,
I am following this document to install outline on a local server. https://docs.getoutline.com/s/hosting/doc/from-source-BlBxrNzMIP

OS: RHEL v9.4

I have installed all the dependencies to their latest levels:
node.js: v22.5.1
yarn: 1.22.22
postgres: will be connecting  to existing v12 server
redis-server: v=6.2.7

I have downloaded the outline-main.zip and unzipped to a local /home/outlineuser directory

modified the .env.example file and called it outline.env

When I run the command `yarn install --frozen-lockfile && yarn build` I get the following error:
```
yarn install v1.22.22
[1/5] Validating package.json...
error outline@0.78.0: The engine "node" is incompatible with this module. Expected version ">= 18 <=20". Got "22.5.1"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```
If I run yarn install --frozen-lockfile separately, I get the following:
```
yarn install v1.22.22
info No lockfile found.
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
Done in 0.05s.
```
when I run yarn build, i get the following error:
```
yarn run v1.22.22
error Command "build" not found.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
look on the web, it seems yarn build is short for yarn run build: https://stackoverflow.com/questions/54693223/what-does-yarn-build-command-do-are-npm-build-and-yarn-build-similar

It just seems like I am missing a file that yarn build is suppose to run against

from the same line above, one person states:
```
build is a command which can be specified in your package.json file on the scripts property. See the example below.

{
    "name": "mypackage",
    "version": "0.1.0",
    "scripts": {
       "build": "webpack --config webpack.dev.js"
    }
}

```
my package.json file in /home/outlineuser is
cat package.json
```
{
  "name": "outlineuser",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT"
}
```
not sure where to go from here, any assistance would be appreciated.

thank you

**Comment by tommoor** [comment]
The error is detailed in the output:

`error outline@0.78.0: The engine "node" is incompatible with this module. Expected version ">= 18 <=20". Got "22.5.1"`

Your version of node is too high
